pv211_utils/evaluation_metrics.py

```diff
@@ -0,0 +1,338 @@
+        System to be evaluated.
+    queries : OrderedDict
+        Queries to be searched.
+    judgements : Set[JudgementBase]
+        Judgements.
+    k : int
+        Parameter defining evaluation depth.
+    num_processes : int
+        Parallelization parameter defining number of processes to be used to run the evaluation.
+
+    Returns:
+    -------
+    float
+        Mean precision score from interval [0, 1].
+    """
+    manager = Manager()
+    mp_score_lock = manager.Lock()
+    mp_score = manager.Value('f', 0.0)
+
+    worker_precision = partial(_calc_precision, system,
+                               _judgements_obj_to_id(judgements),
+                               k, mp_score_lock, mp_score)
+
+    process_pool = Pool(processes=num_processes)
+    process_pool.map(worker_precision, list(queries.values()))
+
+    mp_score.value /= len(queries)
+
+    return mp_score.value
+
+
+def mean_recall(system: IRSystemBase, queries: OrderedDict,
+                judgements: Set[JudgementBase], k: int, num_processes: int) -> float:
+    """Evaluate system for given queries and judgements with mean recall metric.
+    Where first k documents will be used in evaluation.
+
+    Args:
+    ----
+    system : IRSystemBase
+        System to be evaluated.
+    queries : OrderedDict
+        Queries to be searched.
+    judgements : Set[JudgementBase]
+        Judgements.
+    k : int
+        Parameter defining evaluation depth.
+    num_processes : int
+        Parallelization parameter defining number of processes to be used to run the evaluation.
+
+    Returns:
+    -------
+    float
+        Mean recall score from interval [0, 1].
+    """
+    manager = Manager()
+    mr_score_lock = manager.Lock()
+    mr_score = manager.Value('f', 0.0)
+
+    worker_recall = partial(_calc_recall, system,
+                            _judgements_obj_to_id(judgements),
+                            k, mr_score_lock, mr_score)
+
+    process_pool = Pool(processes=num_processes)
+    process_pool.map(worker_recall, list(queries.values()))
+
+    mr_score.value /= len(queries)
+
+    return mr_score.value
+
+
+def normalized_discounted_cumulative_gain(system: IRSystemBase,
+                                          queries: OrderedDict,
+                                          judgements: Set[JudgementBase],
+                                          k: int, num_processes: int) -> float:
+    """Evaluate system for given queries and judgements with normalized
+    discounted cumulative gain metric. Where first k documents will be used in evaluation.
+
+    Args:
+    ----
+    system : IRSystemBase
+        System to be evaluated.
+    queries : OrderedDict
+        Queries to be searched.
+    judgements : Set[JudgementBase]
+        Judgements.
+    k : int
+        Parameter defining evaluation depth.
+    num_processes : int
+        Parallelization parameter defining number of processes to be used to run the evaluation.
+
+    Returns:
```

Comment:
The pv211-utils library uses [the NumPy docstring style guide][1]. `Args:` should be `Arguments`, `Returns:` should be `Returns`, etc. Here and everywhere.

 [1]: https://numpydoc.readthedocs.io/en/latest/format.html

Reply:
todo

Reply:
Fixed in 40d56a9 with the following snippet:

```
find -name '*.py' -exec sed -i -e s/Args:/Arguments/ -e s/Returns:/Returns/ {} +
```